bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java

```diff
@@ -258,6 +258,9 @@ private boolean updateRelayStatus(ShellySettingsStatus status, @Nullable Shelly2
 
         ShellySettingsRelay rstatus;
         ShellyShortStatusRelay sr;
+        if (rs.id == null) { // firmware 1.6.1 returns id = null!
```

Comment:
What will be the difference between using the (hardcoded) provided `id` and the one that might be provided in the DTO? Could we use the hardcoded one always, or is there some drawback to that?

Reply:
I don't like that id. This handling is related to a firmware bug = work around. If I saw correctly 1.7.x returns the component id as expected